src/main/java/flow/AbstractTripleRematch.java

```diff
@@ -0,0 +1,225 @@
+				Triple triple = sortedFileFacade.pop();
+				while (triple != null) {
+					rematchedTriple(triple);
+					triple = sortedFileFacade.pop();
+				}
+			}
+			onFinished();
+		} catch (final IOException e) {
+			throw new MetafactureException("Error merging temp files", e);
+		} finally {
+			if (queue != null) {
+				for (final SortedTripleFileFacade sortedFileFacade : queue) {
+					sortedFileFacade.close();
+				}
+			}
+		}
+		MemoryWarningSystem.removeListener(this);
+	}
+
+	/**
+	 */
+	protected abstract void identifyRekeyCases();
+
+	/**
+	 */
+	protected void onFinished() {
+		// nothing to do
+	}
+
+	@Override
+	public final void onResetStream() {
+		buffer.clear();
+		for (final File file : tempFiles) {
+			if (file.exists()) {
+				file.delete();
+			}
+		}
+		tempFiles.clear();
+	}
+
+	/**
+	 * @param aSubject defines the subject of all matching triples are to be
+	 *          returned.
+	 * @return an ArrayList of triples matching
+	 * @throws IOException if
+	 */
+	protected List<Triple> getTriplesBySubject(String aSubject)
+			throws IOException {
+		List<Triple> matchingTriples = new ArrayList<>();
+		nextBatch();
+		List<SortedTripleFileFacade> queue = getTempFilesAsFacadeList();
+		for (SortedTripleFileFacade sortedFileFacade : queue) {
+			Triple triple = sortedFileFacade.pop();
+			while (triple != null) {
+				if (triple.getSubject().equals(aSubject)) {
+					matchingTriples.add(triple);
+				}
+				triple = sortedFileFacade.pop();
+			}
+		}
+		return matchingTriples;
+	}
+
+	/**
+	 * @param aPredicate defines the predicate of all matching triples are to be
+	 *          returned.
+	 * @return an ArrayList of triples matching
+	 * @throws IOException if
+	 */
+	protected List<Triple> getTriplesByPredicate(String aPredicate)
+			throws IOException {
+		List<Triple> matchingTriples = new ArrayList<>();
+		nextBatch();
+		final List<SortedTripleFileFacade> queue = getTempFilesAsFacadeList();
+		for (SortedTripleFileFacade sortedFileFacade : queue) {
+			Triple triple = sortedFileFacade.pop();
+			while (triple != null) {
+				if (triple.getPredicate().equals(aPredicate)) {
+					matchingTriples.add(triple);
+				}
+				triple = sortedFileFacade.pop();
+			}
+		}
+		return matchingTriples;
+	}
+
+	/**
+	 * @param aObject defines the object of all matching triples are to be
+	 *          returned.
+	 * @return an ArrayList of triples matching
+	 * @throws IOException if
+	 */
+	protected List<Triple> getTriplesByObject(String aObject) throws IOException {
+		List<Triple> matchingTriples = new ArrayList<>();
+		nextBatch();
+		List<SortedTripleFileFacade> queue = getTempFilesAsFacadeList();
+		for (SortedTripleFileFacade sortedFileFacade : queue) {
+			Triple triple = sortedFileFacade.pop();
+			while (triple != null) {
+				if (triple.getObject().equals(aObject)) {
+					matchingTriples.add(triple);
+				}
+				triple = sortedFileFacade.pop();
+			}
+		}
+		return matchingTriples;
+	}
+
```

Comment:
These last two methods are identically except for one line. (This is just a remark. No pressure to change anything ;) )